### ISSUE TITLE
Task: Implement MIRI MRS flux calibration

### ISSUE THREAD

**Issue by hbushouse** [STScI-JWST/jwst#611]
All spectroscopic instrument modes (so far) define a wavelength-dependent spectral response function that's stored in a PHOTOM reference file and loaded by the photom pipeline step into the RELSENS table extension of the science product. The RELSENS data are then used during the extract_1d step to convert observed count rates into absolute fluxes. Flux calibration for the MIRI MRS mode, on the other hand, is being provided as a 2D array of flux conversion factors that are to be applied pixel-by-pixel to the 2D stage of MRS slice images, to convert them from count rates to fluxes.

So at minimum we need to define a new PHOTOM reference file format for the MRS 2D flux array and modify the PHOTOM step to attach this array to the science product, rather than the traditional RELSENS table extension.

We currently believe that the flux-cal information will be applied to the science data during the cube_build process, such that the output cubes from that step will be in units of physical fluxes.

**Comment by hbushouse** [comment]
Copies of the new MRS photom/flux-cal reference files and documentation are available in:
file:///grp/jwst/wit/miri/CDP-6/CALSPEC2_MRS/absolute_flux_calibration/PHOTOM_Jan2017/

There is one ref file per sub-band (short, medium, long) for each detector, for a total of 6 reference files.

**Comment by hbushouse** [comment]
Discussion of this approach with both the MIRI MRS and NIRSpec IFU reps has led to the realization that the NIRSpec IFU group is expecting a similar approach for their calibration, which was not known until now. The NIRSpec IFU mode will still use a photom reference file that defines a simple 1D spectral response function as a function of wavelength, but instead of just attaching that 1D RSRF to the science data as is currently done, they expect it to be used to create, on the fly, a full 2D array that gives the response of each pixel, based on its wavelength. This 2D response array then gets attached to the science product, as is the case with MIRI MRS, and is also applied to the 2D unrectified data on its way into the cube_build routine, so that in both cases cube_build is working with calibrated data that have the effects of varying pixel sizes removed. A separate issue will be filed for the NIRSpec IFU work.

**Comment by philhodge** [comment]
If I understand correctly, this means that the extract_1d step should not even attempt to apply flux calibration for any IFU data.  The net and background columns will be in units of flux density, and the net will be the same as the flux.

**Comment by hbushouse** [comment]
That's my understanding as well. The cubes coming out of cube_build should already be in calibrated units, for both instruments.

**Comment by hbushouse** [comment]
The new MRS photom reference files have the following structure:

```
HDU  Ext type   Datatype  Size
Pri                                 null data array; header contains PHOTMJSR, PHOTUJA2 keywords
SCI     IMAGE   float    1032x1024  units of DN/sec/mJy/pixel
ERR     IMAGE   float    1032x1024  units of DN/sec/mJy/pixel
DQ      IMAGE   int      1032x1024
DQ_DEF  TABLE
PIXSIZ  IMAGE   float    1032x1024  units of arcsec^2
```

The SCI data of the reference file are to be divided into the countrate values of a science image, yielding pixel values in units of mJy/pixel. Further division by the PIXSIZ array yields science image pixels in units of mJy/arcsec^2.

**Comment by philhodge** [comment]
On 02/10/2017 03:16 PM, Howard Bushouse wrote:
> |SCI IMAGE float 1032x1024 units of DN/sec/mJy/pixel|

Just a suggestion:  I'm confident that (in this case) the unit string 
above means (DN/s) / (mJy/pixel), but without parentheses it really is 
ambiguous.  In Python or C, for example, without parentheses it would 
mean DN / (sec * mJy * pixel).  Note that (a / b) / c is not the same as 
a / (b / c).

**Comment by hbushouse** [comment]
Fixed by #694 and #698. Closing.
